[Docs] manual instrumentation (#126)
signalfx/splunk-otel-dotnet: docs/manual-instrumentation.md

> # Manually instrument a .NET application

Author: clarified in https://github.com/signalfx/splunk-otel-dotnet/pull/126/commits/8271315acf4cab6bd19f7fa0328e5e338bc7c28c
`OTEL_DOTNET_AUTO_TRACES_ADDITIONAL_SOURCES` needs to be set if manual spans are created using `System.Diagnostics` API's (linked documentation from upstream mentions how to set `OTEL_DOTNET_AUTO_TRACES_ADDITIONAL_SOURCES` )
I added a separate section for custom metrics, and mentioned `OTEL_DOTNET_AUTO_METRICS_ADDITIONAL_SOURCES` there.

> The auto-instrumentation provides a base you can build on by adding your own

Reviewer: ```suggestion
The automatic instrumentation provides a base you can build upon by adding your own
```

custom instrumentation. By using both instrumentation approaches,

> you'll be able to present a more detailed representation of the logic

Reviewer: ```suggestion
you can produce a more detailed representation of the logic
```

and functionality of your application, clients, and framework.

## Instrument using System.Diagnostics API

> For the list of steps required, see [link](https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/blob/main/docs/manual-instrumentation.md).

Reviewer: ```suggestion
For the list of steps required, see the [upstream documentation](https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/blob/main/docs/manual-instrumentation.md).
```

Reviewer: Any step that would be custom for our distribution?

Author: No custom steps needed for our distribution.

> ## Instrument using OpenTracing API

Reviewer 2: I think we should add a hyperlink in `migrating.md` to this header.

Reviewer: Couldn’t we move the section to the migration section? It feels a bit strange to me that someone would use OpenTracing when a new Otel rostro is available. At the very minimum we should explain why would someone use OT here.

1. Add the `OpenTelemetry.Shims.OpenTracing` dependency to your project:

    ```xml
    <PackageReference Include="OpenTelemetry.Shims.OpenTracing" Version="1.0.0-rc9.10" />

>     ```

Reviewer 2: Is it not enough to reference `OpenTracing`?

Reviewer 2: So we need also to reference `System.Diagnostics.DiagnosticSource` `7.0.0`.

Author: changed in https://github.com/signalfx/splunk-otel-dotnet/pull/126/commits/8271315acf4cab6bd19f7fa0328e5e338bc7c28c
In version 0.5.0 of automatic instrumentation, no additional span is created (tags are added to existing span).
This seems to be fixed in latest versions of automatic instrumentation.

> 1. Set `OTEL_DOTNET_AUTO_OPENTRACING_ENABLED` environment variable to `true`

Reviewer: ```suggestion
1. Set the `OTEL_DOTNET_AUTO_OPENTRACING_ENABLED` environment variable to `true`.
```
